staging/src/k8s.io/client-go/tools/cache/reflector_test.go

```diff
@@ -358,6 +361,54 @@ func TestReflectorListAndWatchWithErrors(t *testing.T) {
 	}
 }
 
+func TestReflectorListAndWatchInitConnBackoff(t *testing.T) {
+	table := []struct {
+		numConnFails float64
+	}{
+		{3.0},
+		{6.0},
+	}
+	for _, test := range table {
+		t.Run(fmt.Sprintf("%d connection failures takes at least %d ms", int(test.numConnFails), int(math.Pow(2, test.numConnFails))),
+			func(t *testing.T) {
+				stopCh := make(chan struct{})
+				connFails := test.numConnFails
+				lw := &testLW{
+					WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+						if connFails > 0 {
+							connFails--
+							return nil, syscall.ECONNREFUSED
+						}
+						close(stopCh)
+						return watch.NewFake(), nil
+					},
+					ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+						return &v1.PodList{ListMeta: metav1.ListMeta{ResourceVersion: "1"}}, nil
+					},
+				}
+				r := &Reflector{
+					name:                   "test-reflector",
+					listerWatcher:          lw,
+					store:                  NewFIFO(MetaNamespaceKeyFunc),
+					initConnBackoffManager: wait.NewExponentialBackoffManager(time.Millisecond, 50*time.Millisecond, 100*time.Millisecond, 2.0, 1.0, &clock.RealClock{}),
+					clock:                  &clock.RealClock{},
```

Comment:
If you use a fake clock, you can make the test have a short wall time while still testing many iterations.